src/app/models/function.rb

```diff
@@ -31,7 +31,8 @@ def find_first_function(ast)
 
   def function_name
     self.name = find_first_function(RubyVM::AbstractSyntaxTree.parse(code))
-  rescue ErrorClass
-    self.name = nil
+  rescue SyntaxError
+    errors.add(:has_syntax_error, 'code has syntax error')
+    throw(:abort)
```

Comment:
[R] we don't need to throw, if the errors are not empty then the validate fails right?

Reply:
Without aborting here the user would get two duplicate errors, the first one is the syntax error and the second one is `name` would be empty.

Reply:
it's alright to have this duplication, I accept it instead of having a condition on the validation on throwing here